### PR TITLE
fix #34: return Result from request query

### LINKED ISSUE #34: Request Query should return result
https://github.com/SeunLanLege/arc-reactor/blob/master/src/core/request.rs#L107-L117

currently while parsing the request query, we're ignoring the errors. This isn't the appropriate approach

`request.query()`

should return a result with an error of QueryParseError

```rust
enum QueryParseError {
  SerdeError(serde_json::Error),
  ParseError(queryst_prime::Error),
  None
}
```

### PR DESCRIPTION
**Pull request by TNkemdilim**

### CODE REVIEW

**Comment by SeunLanLege** [comment]
beautiful work, can you add `QueryParseError` to the library exports so i can merge this.

**Comment by SeunLanLege** [comment]
also check out src/proto/convert and implement the `From` trait for `QueryParseError`

take a look at the `JsonError` implementation for examples.

**Comment by TNkemdilim** [comment]
Should custom messages be used (which I believe is what should be done)?

For:
`queryst_prime::ParseError`, it should output ***failed to deserialize data***
`serde_json::Error`, it should output ***failed to parse query data***

File: src/proto/convert.rs [hunk 1]
```diff
@@ -81,3 +81,30 @@ impl From<JsonError> for Response {
...
+			QueryParseError::None => {
+				let json = json!({
+					"error": "query data was empty",
+				});
+				res().with_body(to_string(&json).unwrap())
+			}
+
+			QueryParseError::ParseError(_e) => {
+				let json = json!({
+					"error": "failed to parse query data",
```
Comment (SeunLanLege) [inline src/proto/convert.rs]: I think we should report the parser error back to the user

File: src/proto/convert.rs [hunk 2]
```diff
@@ -81,3 +81,30 @@ impl From<JsonError> for Response {
...
+				});
+				res().with_body(to_string(&json).unwrap())
+			}
+
+			QueryParseError::SerdeError(_e) => {
+				let json = json!({
+					"error": "failed to deserialize data",
+				});
+				res().with_body(to_string(&json).unwrap())
+			}
```
Comment (SeunLanLege) [inline src/proto/convert.rs]: same here we should report the serde error to the user.

**Comment by SeunLanLege** [comment]
take a look at https://github.com/SeunLanLege/arc-reactor/blob/master/src/proto/convert.rs#L75-L80

**Comment by TNkemdilim** [comment]
I thought about that also. Will fix asap.

**Pull request merged**